src/Symfony/Component/Config/Tests/Definition/ScalarNodeTest.php

```diff
@@ -57,4 +57,51 @@ public function getInvalidValues()
             array(new \stdClass()),
         );
     }
+
+    /**
+     * @dataProvider getValidNonEmptyValues
+     *
+     * @param mixed $value
+     */
+    public function testValidNonEmptyValues($value)
+    {
+        $node = new ScalarNode('test');
+        $node->setAllowEmptyValue(false);
+
+        $this->assertSame($value, $node->finalize($value));
+    }
+
+    public function getValidNonEmptyValues()
+    {
+        return array(
```

Comment:
Well shouldn't that value be there as well ?

Maybe I'm wrong, I just wanted to make sure you didn't accidentally miss it :)